client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -193,6 +193,24 @@ public ShuffleClientImpl(String appUniqueId, CelebornConf conf, UserIdentifier u
             "celeborn-shuffle-split", pushSplitPartitionThreads, 60);
     reviveManager = new ReviveManager(this, conf);
 
+    if (!isDriver) {
+      heartbeater =
+          ThreadUtils.newDaemonSingleThreadScheduledExecutor("lifecycleManager-heartbeater");
+      heartbeater.scheduleAtFixedRate(
+          () -> {
+            PbHeartbeatFromClientResponse resp =
+                lifecycleManagerRef.askSync(
+                    HeartbeatFromClient$.MODULE$.apply(reducePartitionMap.keySet()),
+                    ClassTag$.MODULE$.apply(PbHeartbeatFromClientResponse.class));
+            List<Integer> unkownShuffleIds = resp.getUnkownShuffleIdList();
+            for (int i = 0; i < unkownShuffleIds.size(); i++) {
+              unregisterShuffle(unkownShuffleIds.get(i), false);
+            }
+          },
+          60,
```

Comment:
configurable?

Reply:
I feel it unnecessary to make it configurable...

Reply:
I don't oppose your decision. Generally, we prefer to make it configurable, especially we can use 0 to disable the feature to restore the old behavior in case the new feature introduces other expected impacts.

Reply:
done, thanks for your explanation.